api/api.go

```diff
@@ -437,6 +441,62 @@ type CloudBackupSchedEnumerateResponse struct {
 	Schedules map[string]CloudBackupScheduleInfo
 }
 
+type Operation string
+
+const (
+	MigrateCluster     = Operation("MigrateCluster")
+	MigrateVolume      = Operation("MigrateVolume")
+	MigrateVolumeGroup = Operation("MigrateVolumeGroup")
+)
+
+type MigrateStatus string
+
+const (
+	MigrateStatusQueued       = MigrateStatus("Queued")
+	MigrateStatusInitializing = MigrateStatus("Initializing")
+	MigrateStatusInProgress   = MigrateStatus("InProgress")
+	MigrateStatusFailed       = MigrateStatus("Failed")
+	MigrateStatusComplete     = MigrateStatus("Complete")
+)
+
+type MigrateStage string
+
+const (
+	MigrateStageBackup       = MigrateStage("Backup")
+	MigrateStageRestore      = MigrateStage("Restore")
+	MigrateStageVolumeUpdate = MigrateStage("VolumeUpdate")
+	MigrateStageDone         = MigrateStage("Done")
+)
+
+type CloudMigrateInfo struct {
+	ClusterId       string
+	LocalVolumeId   string
+	LocalVolumeName string
+	RemoteVolumeId  string
+	CloudSnapId     string
+	CurrentStage    MigrateStage
+	Status          MigrateStatus
+	LastUpdate      time.Time
+	LastSuccess     time.Time
+}
+
+type CloudMigrateStartRequest struct {
+	Operation Operation
+	ClusterId string
+	// Depending on Operation type this can be VolumeID or VolumeGroupID
+	TargetId string
```

Comment:
Change this to be volume_target_id

Reply:
This can be volume of group id. That's why it is called target id

Reply:
As the author of the code this makes sense, but as a reader it is difficult to get that information from `TargetId`. As a reader, it would make it easier to have variables be self-documenting.

Reply:
There is a comment to clarify it. volume_target_id doesn't make sense since it can also be a group id.